core/src/main/java/se/tre/freki/core/MetaClient.java

```diff
@@ -122,14 +122,15 @@ public void deleteAnnotation(final Annotation note) {
    * @param id The id of the label to lookup the meta information about
    * @return A future that on completion contains the meta object
    */
-  public ListenableFuture<LabelMeta> getLabelMeta(final LabelType type,
-                                                  final LabelId id) {
+  public ListenableFuture<Optional<LabelMeta>> getLabelMeta(final LabelType type,
+                                                            final LabelId id) {
     // Verify that the identifier exists before fetching the meta object. The future that
     // #getLabelName returns will contain an exception if it does not exist.
     return transform(labelClient.getLabelName(type, id),
```

Comment:
Checking this is probably redundant now when we return an `Optional<LabelMeta>`?